### PR TITLE
Remove jQuery AJAX from the comment edit box

### PR DESCRIPTION
**Pull request by yardenshoham**
- Removed all jQuery AJAX calls and replaced with our fetch wrapper
- Tested the file addition and removal functionality and it works as before

# Demo using `fetch` instead of jQuery AJAX
demo

### CODE REVIEW

File: web_src/js/features/repo-legacy.js [hunk 1]
```diff
@@ -355,23 +355,28 @@ async function onEditContent(event) {
...
           $(`#${file.uuid}`).remove();
           if ($dropzone.attr('data-remove-url') && !fileUuidDict[file.uuid].submitted) {
-            $.post($dropzone.attr('data-remove-url'), {
-              file: file.uuid,
-              _csrf: csrfToken,
-            });
+            try {
+              const params = new URLSearchParams();
+              params.append('file', file.uuid);
+              await POST($dropzone.attr('data-remove-url'), {data: params});
```
Comment (silverwind) [inline web_src/js/features/repo-legacy.js]:
```suggestion
              await POST($dropzone.attr('data-remove-url'), {data: new URLSearchParams({file: file.uuid})});
```

I think this is the third time I make this suggestion now. When will you learn? 😆

Comment (yardenshoham) [reply web_src/js/features/repo-legacy.js]: Oof I promise it's the last time :)

File: web_src/js/features/repo-legacy.js [hunk 2]
```diff
@@ -406,22 +413,25 @@ async function onEditContent(event) {
...
-      context: $editContentZone.attr('data-context'),
-      files: dz.files.map((file) => file.uuid),
-    }, (data) => {
+
+    try {
+      const params = new URLSearchParams({
+        content: comboMarkdownEditor.value(),
+        context: $editContentZone.attr('data-context')
+      });
+      for (const file of dz.files) params.append('files[]', file.uuid);
```
Comment (silverwind) [inline web_src/js/features/repo-legacy.js:423]: TIL that URLSearchParams can have duplicate keys 😆

**Review by silverwind — APPROVED** [review]
ok if it works

File: web_src/js/features/repo-legacy.js [hunk 3]
```diff
@@ -406,22 +411,25 @@ async function onEditContent(event) {
...
-      _csrf: csrfToken,
-      content: comboMarkdownEditor.value(),
-      context: $editContentZone.attr('data-context'),
-      files: dz.files.map((file) => file.uuid),
-    }, (data) => {
+
+    try {
+      const params = new URLSearchParams({
+        content: comboMarkdownEditor.value(),
+        context: $editContentZone.attr('data-context')
```
Comment (silverwind) [inline web_src/js/features/repo-legacy.js]:
```suggestion
        context: $editContentZone.attr('data-context'),
```

**Review by delvh — APPROVED** [review]

**Pull request merged**